pkg/patterns/application/example_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package application_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/patterns/application"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+)
+
+// This example creates a simple application Controller that is configured for ReplicaSets and Pods.
+//
+// * Create a new application for ReplicaSets that manages Pods owned by the ReplicaSet and calls into
+// ReplicaSetController.
+//
+// * Start the application.
+func ExampleApplication() {
+	a, err := application.
+		BuilderFor(&appsv1.ReplicaSet{}).     // ReplicaSet is the Application API
+		Owns(&corev1.Pod{}).                  // ReplicaSet owns Pods created by it
+		WithHandler(&ReplicaSetController{}). // ReplicaSet and Pod events (create/update/del) trigger the handler
+		Build()                               // Build
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Fatal(a.Start(signals.SetupSignalHandler()))
+}
+
+// ReplicaSetController is a simple Controller example implementation.
+type ReplicaSetController struct {
+	client.Client
+}
+
+// Implement the business logic:
+// This function will be called when there is a change to a ReplicaSet or a Pod with an OwnerReference
+// to a ReplicaSet.
+//
+// * Read the ReplicaSet
+// * Read the Pods
+// * Set a Label on the ReplicaSet with the Pod count
+func (a *ReplicaSetController) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	// Read the ReplicaSet
+	rs := &appsv1.ReplicaSet{}
+	err := a.Get(context.TODO(), req.NamespacedName, rs)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// List the Pods matching the PodTemplate Labels
+	pods := &corev1.PodList{}
+	err = a.List(context.TODO(),
+		&client.ListOptions{
```

Comment:
I believe we have a builder for this (if we don't have one for the namespace, we should), so you should be able to write something like

`client.InNamespace(req.Namespace).WithLabels(rs.Spec.Template.Labels)`, which is a bit easier to read and doesn't require pulling out `labels.SelectorFromSet`.

Reply:
Done